# EDA Algerian Forest Fires 

## 1) Problem Statement
This project, focuses on whether certain weather features could predict forest fires in some regions in Algeria using few Classification algorithms.

## 2) Data Source
https://archive.ics.uci.edu/dataset/547/algerian+forest+fires+dataset

## 3) About the dataset

The dataset includes 246 observations that is a regroup of data of two regions of Algeria,namely the Bejaia region located in the northeast of Algeria and the Sidi Bel-abbes region located in the northwest of Algeria.

There are 123 observations for each region. 

The period considered here for observation is from June 2012 to September 2012. 
The dataset includes 11 input features (day, month,	year together considered as date) and 1 output attribue (Classes).

## 4) Import Libraries

In [65]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
from warnings import filterwarnings
filterwarnings('ignore')

## 5) Preparing the dataset

In [62]:
df = pd.read_csv('Algerian_forest_fires_dataset_UPDATE (6) (1).csv', header=1)
df.columns = df.columns.str.strip()

In [63]:
df.head()

day month  year Temperature  RH  Ws  Rain  FFMC  DMC    DC  ISI  BUI  FWI  \
0  01    06  2012          29  57  18     0  65.7  3.4   7.6  1.3  3.4  0.5   
1  02    06  2012          29  61  13   1.3  64.4  4.1   7.6    1  3.9  0.4   
2  03    06  2012          26  82  22  13.1  47.1  2.5   7.1  0.3  2.7  0.1   
3  04    06  2012          25  89  13   2.5  28.6  1.3   6.9    0  1.7    0   
4  05    06  2012          27  77  16     0  64.8    3  14.2  1.2  3.9  0.5   

       Classes  
0  not fire     
1  not fire     
2  not fire     
3  not fire     
4  not fire

## 6) Attribute Information

1. Date : (DD/MM/YYYY) Day, month ('june' to 'september'), year (2012) - Date of the data observations 
2. Temp : temperature noon (temperature max)  in Celsius degrees: 22 to 42
3. RH : Relative Humidity in %: 21 to 90 
4. Ws :Wind speed in km/h: 6 to 29 
5. Rain: total day in mm: 0 to 16.8
FWI Components  
6. Fine Fuel Moisture Code (FFMC) index from the FWI system: 28.6 to 92.5 
7. Duff Moisture Code (DMC) index from the FWI system: 1.1 to 65.9 
8. Drought Code (DC) index from the FWI system:  7 to 220.4
9. Initial Spread Index (ISI) index from the FWI system: 0 to 18.5 
10. Buildup Index (BUI) index from the FWI system: 1.1 to 68
11. Fire Weather Index (FWI) Index: 0 to 31.1
12. Classes: two classes, namely   “Fire” and “not Fire”

## 7) Check if there are any missing values

In [66]:
df.isnull().sum()

day            0
month          1
year           1
Temperature    1
RH             1
Ws             1
Rain           1
FFMC           1
DMC            1
DC             1
ISI            1
BUI            1
FWI            1
Classes        2
dtype: int64

In [67]:
df[df.isnull().any(axis=1)]

day month  year Temperature   RH   Ws Rain  \
122  Sidi-Bel Abbes Region Dataset   NaN   NaN         NaN  NaN  NaN  NaN   
167                             14    07  2012          37   37   18  0.2   

     FFMC   DMC      DC   ISI   BUI      FWI Classes  
122   NaN   NaN     NaN   NaN   NaN      NaN     NaN  
167  88.9  12.9  14.6 9  12.5  10.4  fire        NaN

#### There are no such missing records observed except for one record in fearure 'Classes'.

#### So till the index 122, the region was Bejaia and from index 123, the region was Sidi-Bel Abbes.

## 8) Handling missing values

In [101]:
Data['Classes'].fillna(df['Classes'].mode()[0], inplace=True)

#### We can seperate the dataset into two subgroups on the basis of region and then rejoin the subgroups and drop the row index 122.

In [68]:
df1=df.iloc[:123]
df2=df.iloc[123:]

In [70]:
df1['Region'] = 'Bejaia'
df2['Region'] = 'Sidi Bel-abbes'

In [71]:
df1.head()

day month  year Temperature  RH  Ws  Rain  FFMC  DMC    DC  ISI  BUI  FWI  \
0  01    06  2012          29  57  18     0  65.7  3.4   7.6  1.3  3.4  0.5   
1  02    06  2012          29  61  13   1.3  64.4  4.1   7.6    1  3.9  0.4   
2  03    06  2012          26  82  22  13.1  47.1  2.5   7.1  0.3  2.7  0.1   
3  04    06  2012          25  89  13   2.5  28.6  1.3   6.9    0  1.7    0   
4  05    06  2012          27  77  16     0  64.8    3  14.2  1.2  3.9  0.5   

       Classes  Region  
0  not fire     Bejaia  
1  not fire     Bejaia  
2  not fire     Bejaia  
3  not fire     Bejaia  
4  not fire     Bejaia

In [72]:
df2.head()

day  month  year  Temperature   RH   Ws   Rain  FFMC  DMC    DC  ISI  \
123  day  month  year  Temperature   RH   Ws  Rain   FFMC  DMC    DC  ISI   
124   01     06  2012           32   71   12    0.7  57.1  2.5   8.2  0.6   
125   02     06  2012           30   73   13      4  55.7  2.7   7.8  0.6   
126   03     06  2012           29   80   14      2  48.7  2.2   7.6  0.3   
127   04     06  2012           30   64   14      0  79.4  5.2  15.4  2.2   

     BUI  FWI      Classes          Region  
123  BUI  FWI    Classes    Sidi Bel-abbes  
124  2.8  0.2  not fire     Sidi Bel-abbes  
125  2.9  0.2  not fire     Sidi Bel-abbes  
126  2.6  0.1  not fire     Sidi Bel-abbes  
127  5.6    1  not fire     Sidi Bel-abbes

In [73]:
Data = pd.concat([df1, df2])
Data.drop(122, inplace=True)

In [99]:
Data

day month  year Temperature  RH  Ws  Rain  FFMC  DMC    DC  ISI   BUI  \
0    01    06  2012          29  57  18     0  65.7  3.4   7.6  1.3   3.4   
1    02    06  2012          29  61  13   1.3  64.4  4.1   7.6    1   3.9   
2    03    06  2012          26  82  22  13.1  47.1  2.5   7.1  0.3   2.7   
3    04    06  2012          25  89  13   2.5  28.6  1.3   6.9    0   1.7   
4    05    06  2012          27  77  16     0  64.8    3  14.2  1.2   3.9   
..   ..   ...   ...         ...  ..  ..   ...   ...  ...   ...  ...   ...   
241  26    09  2012          30  65  14     0  85.4   16  44.5  4.5  16.9   
242  27    09  2012          28  87  15   4.4  41.1  6.5     8  0.1   6.2   
243  28    09  2012          27  87  29   0.5  45.9  3.5   7.9  0.4   3.4   
244  29    09  2012          24  54  18   0.1  79.7  4.3  15.2  1.7   5.1   
245  30    09  2012          24  64  15   0.2  67.3  3.8  16.5  1.2   4.8   

     FWI       Classes          Region  
0    0.5   not fire             Bejaia  
1    0.4   not fire             Bejaia  
2    0.1   not fire             Bejaia  
3      0   not fire             Bejaia  
4    0.5   not fire             Bejaia  
..   ...           ...             ...  
241  6.5       fire     Sidi Bel-abbes  
242    0   not fire     Sidi Bel-abbes  
243  0.2   not fire     Sidi Bel-abbes  
244  0.7   not fire     Sidi Bel-abbes  
245  0.5  not fire      Sidi Bel-abbes  

[245 rows x 15 columns]

In [102]:
Data.isnull().sum()

day            0
month          0
year           0
Temperature    0
RH             0
Ws             0
Rain           0
FFMC           0
DMC            0
DC             0
ISI            0
BUI            0
FWI            0
Classes        0
Region         0
dtype: int64

#### Now there are no missing values in the dataset.

## 9) Check if there are any duplicate date in the dataset.

In [105]:
df.duplicated().sum()

0

#### So, no duplicate data found. 